NOAA-EMC/EVS
# Feature/sref fixes

<b>Note to developers: You must use this PR template!</b>

## Description of Changes
This PR is to fix several issues  listed in the EVS/v2 Additions and Bugfixes document, and some other issues that do not
listed in this document but are general requirements for EVS. The fixed issues include:
1.  In MET/METplus-beta5 environment 
     ALL of 6 items for SREF  listed in the "additions and fixes" are addressed: 
2. Each each MPMD child process writes output files in its own directory for both stats and graphics jobs (No.1)
3. Changed subprocess.Popen to subprocess.run in EVS v2.0 (if they are used at all) (No. 2)
4. Defined $NET prior to $HOMEevs; apply this change across all J-jobs (No. 3)
5. Removes old concatenating log file in stats job, but keep it in plotting job (No. 4)
6. Used the new seeting for CAPE in the conf files (No. 5)
7. Removed the WARNING for reading duplicated APCP_03 fields from the arw member files at f03 in the precip stat job (No. 6)

Following 8~11 are general requirements for EVS components:
8. Removed LOOP_ORDER, LOG_LINE_FORMATi, VERBOSITY, LOG_LEVEL  from conf file
9. Changed past31/90days to last31/90days in the plot filename
10. Added cheching the availability of the input fcst and obsv files before running METplus in ush/script files
11. Added cat log files for all plottting jobs (it was missing in the old version)
12. Adjusted the memory usage for all plotting jobs based on actual memory used
13. Re-named the graphic file names based on the EVS file naming requirement 


## Developer Questions and Checklist
* Is this a high priority PR? If so, why and is there a date it needs to be merged by? 
    Yes,  since all of the issues should be fixed before EVS/v2 implementation deadline
* Do you have any planned upcoming annual leave/PTO? 
    Yes,  11/19 and 11/22 will leave
* Are there any changes needed for when the jobs are supposed to run? 
    No
  
- [y ] The code changes follow [NCO's EE2 Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf).
- [y ] Developer's name is removed throughout the code and have used `${USER}` where necessary throughout the code.
- [y ] References the feature branch for `HOMEevs` are removed from the code.
- [ y] J-Job environment variables, COMIN and COMOUT directories, and output follow what has been [defined](https://docs.google.com/document/d/1JWg_4q80aYmmAoD21GFjp9R9y5-3w7WGM3-0HJk0Pjs/edit#heading=h.7ysbr191vzu4) for EVS.
- [y ] Jobs over 15 minutes in runtime have restart capability.
- [y ] If applicable, changes in the `dev/drivers/scripts` or `dev/modulefiles` have been made in the corresponding `ecf/scripts` and `ecf/defs/evs-nco.def`? 
- [y ] Jobs contain the appropriate file checking and don't run METplus for any missing data.
- [y ] Code is using METplus wrappers structure and not calling MET executables directly.
- [ y] Log is free of any ERRORs or WARNINGs.

## Testing Instructions

Part 1. For the stats generation jobs:

There are 2 independent stat generation jobs:
  jevs_mesoscale_sref_grid2obs_stats.sh
  jevs_mesoscale_sref_precip_stats.sh

 Note: Before running mesoscale_sref_precip_stats.sh, please
    compile to create exec/sref_precip.x or link the EXECevs to /lfs/h2/emc/vpppg/noscrub/emc.vpppg/EVS/exec  in the dev script:
            export EXECevs=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/EVS/exec

Part 2. For the plots generation jobs:

 There are 7 independent plot generation jobs: 
   jevs_mesoscale_sref_cape_last90days_plots.sh   
   jevs_mesoscale_sref_grid2obs_last90days_plots.sh  
   jevs_mesoscale_sref_td2m_last90days_plots.sh
   jevs_mesoscale_sref_cloud_last90days_plots.sh  
   jevs_mesoscale_sref_precip_last90days_plots.sh
   jevs_mesoscale_sref_cnv_last90days_plots.sh    
   jevs_mesoscale_sref_precip_spatial_plots.sh 

 Note: In all of the plotting jobs, the COMIN should be set to /lfs/h2/emc/vpppg/noscrub/emc.vpppg/evs/v2.0
